src/controllers/auth-controller.ts

```diff
@@ -0,0 +1,27 @@
+import {CreateUserRequest, LoginRequest, TypedRequestBody} from "../global";
+import {Response} from "express";
+import {AuthService} from "../services/auth-service";
+
+const authService = new AuthService()
+
+export class AuthController {
+
+    registerUser = async (req: TypedRequestBody<CreateUserRequest>, res: Response) => {
+        const request = req.body
+
+        await authService.registerUser(request);
+
+        res.send("success");
+    };
+
+    login = async(req: TypedRequestBody<LoginRequest>, res: Response) => {
+        const isASuccessfulLogin = await authService.login(req.body);
```

Comment:
Needs error handling, we should also be validating the body conforms and only passing the relevant parameters to our service. 

Working this way too tightly couples the request to the service